contracts/contracts/VanityURL.sol

```diff
@@ -0,0 +1,220 @@
+// SPDX-License-Identifier: CC-BY-NC-4.0
+
+pragma solidity ^0.8.17;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/security/Pausable.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+
+import "./IDC.sol";
+
+contract VanityURL is Ownable, Pausable, ReentrancyGuard {
+    /// @dev DC contract
+    address public dc;
+
+    /// @dev DC TokenId -> Alias Name -> URL
+    mapping(bytes32 => mapping(string => string)) public vanityURLs;
+
+    /// @dev DC TokenId -> Alias Name -> Content Price
+    mapping(bytes32 => mapping(string => uint256)) public vanityURLPrices;
```

Comment:
It has been set but not read. What is the intended use?

Reply:
Yes, vanityURLPrices are not read. It'll be used off-chain to check if he paid enough funds for access when someone is trying to access that vanity url.